File: Agenda/src/main/java/pt/ipcb/mei/aid/agenda/model/AgendaDTO.java
package pt.ipcb.mei.aid.agenda.model;

import java.time.LocalDateTime;

public class AgendaDTO {

    private Long id;
    private Long userId;
    private LocalDateTime inicio;
    private LocalDateTime fim;
    private String titulo;
    private String local;
    private boolean diaInteiro;
    private String recorrencia_rrle;
    private String notas;
    private LocalDateTime datahoraCriacao;
    private LocalDateTime datahoraAlteracao;

    public AgendaDTO() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public LocalDateTime getInicio() {
        return inicio;
    }

    public void setInicio(LocalDateTime inicio) {
        this.inicio = inicio;
    }

    public LocalDateTime getFim() {
        return fim;
    }

    public void setFim(LocalDateTime fim) {
        this.fim = fim;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getLocal() {
        return local;
    }

    public void setLocal(String local) {
        this.local = local;
    }

    public boolean isDiaInteiro() {
        return diaInteiro;
    }

    public void setDiaInteiro(boolean diaInteiro) {
        this.diaInteiro = diaInteiro;
    }

    public String getRecorrencia_rrle() {
        return recorrencia_rrle;
    }

    public void setRecorrencia_rrle(String recorrencia_rrle) {
        this.recorrencia_rrle = recorrencia_rrle;
    }

    public String getNotas() {
        return notas;
    }

    public void setNotas(String notas) {
        this.notas = notas;
    }

    public LocalDateTime getDatahoraCriacao() {
        return datahoraCriacao;
    }

    public void setDatahoraCriacao(LocalDateTime datahoraCriacao) {
        this.datahoraCriacao = datahoraCriacao;
    }

    public LocalDateTime getDatahoraAlteracao() {
        return datahoraAlteracao;
    }

    public void setDatahoraAlteracao(LocalDateTime datahoraAlteracao) {
        this.datahoraAlteracao = datahoraAlteracao;
    }
}

File: mobile/meditrack/lib/model/agenda_dto.dart
class AgendaDTO {
  int? id;
  int userId;
  final DateTime inicio;
  final DateTime? fim;
  final String titulo;
  final String local;
  final bool diaInteiro;
  final String? recorrenciaRrle;
  final String? notas;

  AgendaDTO({
    this.id,
    required this.userId,
    required this.inicio,
    this.fim,
    required this.titulo,
    required this.local,
    required this.diaInteiro,
    this.recorrenciaRrle,
    this.notas,
  });

  factory AgendaDTO.fromJson(Map<String, dynamic> json) {
    return AgendaDTO(
      id: json['id'],
      userId: json['userId'],
      inicio: DateTime.parse(json['inicio']),
      fim: json['fim'] != null ? DateTime.tryParse(json['fim']) : null,
      titulo: json['titulo'] ?? '',
      local: json['local'] ?? '',
      diaInteiro: json['diaInteiro'] ?? false,
      recorrenciaRrle: json['recorrencia_rrle'],
      notas: json['notas'],
    );
  }

  Map<String, dynamic> toJson() {
    final map = {
      'userId': userId,
      'inicio': inicio.toIso8601String(),
      'fim': fim?.toIso8601String(),
      'titulo': titulo,
      'local': local,
      'diaInteiro': diaInteiro,
      'recorrencia_rrle': recorrenciaRrle,
      'notas': notas,
    };

    if (id != null) {
      map['id'] = id;
    }

    return map;
  }
}
